autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -297,6 +301,14 @@ def _generate_retrieve_user_reply(
             messages = self._oai_messages[sender]
         message = messages[-1]
         update_context_case1, update_context_case2 = self._check_update_context(message)
+        if not self._initchat_from_this and not hasattr(self, "problem"):
+            # the first time the agent is called in a group chat
+            message = self.generate_init_message(message.get("content", ""), n_results=self.n_results)
+            self._initchat_from_this = False  # reset to False as the value is changed to True in generate_init_message
+            return True, message
+        elif not self._initchat_from_this and hasattr(self, "problem"):
+            # the agent is called in a group chat and the problem is already set
+            update_context_case1 = True
```

Comment:
Why is this triggered only when `not self_initchat_from_this`? If a chat is initiated from this agent, but the question changed later, shouldn't we also update the context?

Reply:
Right, this is complicated. We have below cases:
1. init chat from raguserproxyagent, in this case, the old implementation is fine.
2. init chat from another userproxyagent, we need to address how to update the problem:
2.1. we take the first message raguserproxyagent received as the problem and we keep it. So if raguserproxyagent is selected again, we update the context. (the current implementation in the PR)
2.2. we replace the problem with the current message sent to raguserproxyagent. So we always retrieve docs for the new problem.

Since in a group chat, raguserproxyagent not always talking to ragassistantagent, the original `update context` logic is not valid.

I think it would be necessary to add some controls to the group chat selector.